component/portal/src/main/java/org/exoplatform/portal/mop/user/UserPortalImpl.java

```diff
@@ -140,15 +140,16 @@ public List<UserNavigation> getNavigations() {
   }
 
   @Override
-  public Collection<UserNode> getNodes(SiteType siteType, Scope scope, UserNodeFilterConfig filterConfig) {
+  public Collection<UserNode> getNodes(SiteType siteType, Scope scope, UserNodeFilterConfig filterConfig, boolean includeGlobal) {
 
     Collection<UserNode> resultUserNodes = new ArrayList<>();
     Set<String> addedUserNodesURI = new HashSet<>();
     for (UserNavigation userNavigation : getNavigations()) {
       SiteKey siteKey = userNavigation.getKey();
       if (siteKey.getType() != siteType
           || (siteType == SiteType.GROUP && siteKey.getName().startsWith(SPACES_SITE_TYPE_PREFIX))
-          || (siteType == SiteType.SPACE && !siteKey.getName().startsWith(SPACES_SITE_TYPE_PREFIX))) {
+          || (siteType == SiteType.SPACE && !siteKey.getName().startsWith(SPACES_SITE_TYPE_PREFIX))
+          || (!includeGlobal && siteKey.getName().equalsIgnoreCase("global"))) {
```

Comment:
WDYT if we modify getNavigations and loadUserNavigation methods instead by changing the condition:

```
if (siteKey.getType() == SiteType.PORTAL
          && StringUtils.isNotBlank(service.getGlobalPortal())
          && !StringUtils.equals(service.getGlobalPortal(), siteKey.getName())) {
        // Add global navigation at the end
        loadUserNavigation(new SiteKey(SiteType.PORTAL, service.getGlobalPortal()));
 }
```

Reply:
in "PORTAL" case, we need to modify getNodes method